Panic when shutting down

```
I0731 10:21:34.212899    1223 chain_manager.go:672] imported 1 block(s) (0 queued 0 ignored) including 0 txs in 4.646155ms. #9711 [893201da / 893201da]
I0731 10:21:34.536329    1223 chain_manager.go:672] imported 1 block(s) (0 queued 0 ignored) including 0 txs in 5.40209ms. #9711 [b0713f41 / b0713f41]
I0731 10:21:34.801144    1223 chain_manager.go:672] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 4.189391ms. #9710 [cb477361 / cb477361]
I0731 10:21:34.828828    1223 chain_manager.go:672] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 4.827608ms. #9711 [893201da / 893201da]
^C
^[[A^CI0731 10:21:41.630326    1223 cmd.go:131] Got interrupt, shutting down...
I0731 10:21:42.272054    1223 chain_manager.go:672] imported 1 block(s) (0 queued 0 ignored) including 0 txs in 7.86914ms. #9712 [0e253dd9 / 0e253dd9]
I0731 10:21:42.741500    1223 chain_manager.go:672] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 5.75718ms. #9711 [893201da / 893201da]
I0731 10:21:42.770617    1223 chain_manager.go:672] imported 0 block(s) (0 queued 1 ignored) including 0 txs in 5.771637ms. #9712 [0e253dd9 / 0e253dd9]
I0731 10:21:43.561187    1223 chain_manager.go:470] Chain manager stopped
I0731 10:21:43.561221    1223 handler.go:165] Stopping ethereum protocol handler...
I0731 10:21:43.561246    1223 handler.go:175] Ethereum protocol handler stopped
I0731 10:21:43.561279    1223 transaction_pool.go:138] TX Pool stopped
I0731 10:21:43.561305    1223 backend.go:695] Automatic pregeneration of ethash DAG OFF (ethash dir: /Users/justin/.ethash)
panic: close of closed channel

goroutine 9083 [running]:
github.com/ethereum/go-ethereum/core.(*ChainManager).Stop(0xc20a0b2a00)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/chain_manager.go:465 +0x2f
github.com/ethereum/go-ethereum/eth.(*Ethereum).Stop(0xc20aa34dc0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/backend.go:609 +0x41
created by github.com/ethereum/go-ethereum/cmd/utils.func·001
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:132 +0x259

goroutine 1 [chan receive]:
main.console(0xc20809e9a0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:437 +0x528
github.com/codegangsta/cli.Command.Run(0x4947130, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4a6ea10, 0x30, 0x4c065f0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:118 +0x104b
github.com/codegangsta/cli.(*App).Run(0xc208172000, 0xc20800a000, 0x4, 0x4, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:177 +0xdf6
main.main()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:340 +0x80

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/Cellar/go/1.4.2/libexec/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/Cellar/go/1.4.2/libexec/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [chan receive]:
github.com/ethereum/go-ethereum/logger/glog.(*loggingT).flushDaemon(0x4f92480)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:897 +0x78
created by github.com/ethereum/go-ethereum/logger/glog.init·1
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:428 +0x77

goroutine 17 [syscall, 5 minutes, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.4.2/libexec/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [select]:
github.com/ethereum/go-ethereum/logger.dispatchLoop()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:81 +0x6d1
created by github.com/ethereum/go-ethereum/logger.init·1
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:60 +0x25

goroutine 9084 [chan receive]:
github.com/ethereum/go-ethereum/logger.sysLoop(0x5187d98, 0xc208194250, 0xc20ac64fc0, 0xc20801e940)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:117 +0x5a
created by github.com/ethereum/go-ethereum/logger.func·001
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:77 +0x176

goroutine 11 [syscall]:
syscall.Syscall(0x3, 0x3, 0xc208196000, 0x8000, 0xc20aa15360, 0x403787a, 0xc208017670)
	/usr/local/Cellar/go/1.4.2/libexec/src/syscall/asm_darwin_amd64.s:20 +0x5
syscall.read(0x3, 0xc208196000, 0x8000, 0x8000, 0xc208017668, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/syscall/zsyscall_darwin_amd64.go:970 +0x6e
syscall.Read(0x3, 0xc208196000, 0x8000, 0x8000, 0x20, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/syscall/syscall_unix.go:136 +0x58
os.(*File).read(0xc20803a128, 0xc208196000, 0x8000, 0x8000, 0x3, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/os/file_unix.go:191 +0x86
os.(*File).Read(0xc20803a128, 0xc208196000, 0x8000, 0x8000, 0x3, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/os/file.go:95 +0x91
io.Copy(0x5186830, 0xc20803a008, 0x5187c58, 0xc20803a128, 0x91f, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:362 +0x1f6
created by main.console
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:406 +0x164

goroutine 13 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc20809ea80)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x21e
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x24c

goroutine 14 [select, 4 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc2081d6160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x48c
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x849

goroutine 15 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc2081d6160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x14a
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x863

goroutine 16 [select, 4 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc2081d6160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x858
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x8a7

goroutine 50 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc2081d6160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x283
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x8c1

goroutine 51 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc2081d6160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x197
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8db

goroutine 52 [select]:
github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).meter(0xc208108000, 0xb2d05e00)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:242 +0xaa5
created by github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).Meter
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:174 +0x537

goroutine 53 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc20809e380)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x21e
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x24c

goroutine 54 [select, 2 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc2081de160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x48c
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x849

goroutine 55 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc2081de160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x14a
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x863

goroutine 56 [select, 2 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc2081de160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x858
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x8a7

goroutine 57 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc2081de160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x283
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x8c1

goroutine 58 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc2081de160)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x197
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8db

goroutine 59 [select]:
github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).meter(0xc2081080c0, 0xb2d05e00)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:242 +0xaa5
created by github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).Meter
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:174 +0x537

goroutine 60 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc20809e620)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x21e
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x24c

goroutine 61 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc2081de580)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x48c
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x849

goroutine 62 [select]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc2081de580)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x14a
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x863

goroutine 63 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc2081de580)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x858
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x8a7

goroutine 64 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc2081de580)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x283
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x8c1

goroutine 65 [select, 5 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc2081de580)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x197
created by github.com/syndtr/goleveldb/leveldb.openDB
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8db

goroutine 66 [select]:
github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).meter(0xc208108180, 0xb2d05e00)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:242 +0xaa5
created by github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).Meter
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethdb/database.go:174 +0x537

goroutine 67 [runnable]:
github.com/ethereum/go-ethereum/core.(*ChainManager).update(0xc20a0b2a00)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/chain_manager.go:760 +0x580
created by github.com/ethereum/go-ethereum/core.NewChainManager
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/chain_manager.go:132 +0x919

goroutine 68 [runnable]:
github.com/ethereum/go-ethereum/core.(*TxPool).eventLoop(0xc20b1857a0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/transaction_pool.go:95 +0x93
created by github.com/ethereum/go-ethereum/core.NewTxPool
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/transaction_pool.go:86 +0x3ce

goroutine 69 [runnable]:
github.com/ethereum/go-ethereum/miner.(*worker).update(0xc20bae0c30)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/miner/worker.go:215 +0x689
created by github.com/ethereum/go-ethereum/miner.newWorker
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/miner/worker.go:135 +0x373

goroutine 70 [chan receive, 5 minutes]:
github.com/ethereum/go-ethereum/miner.(*worker).wait(0xc20bae0c30)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/miner/worker.go:254 +0x92
created by github.com/ethereum/go-ethereum/miner.newWorker
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/miner/worker.go:136 +0x38d

goroutine 72 [select]:
net/http.(*Transport).getConn(0xc208056090, 0xc20a881040, 0x0, 0xc2085f61b0, 0x4, 0xc2085f61b7, 0x12, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:525 +0x608
net/http.(*Transport).RoundTrip(0xc208056090, 0xc20a881040, 0xc20a9996c0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:228 +0x4d4
net/http.send(0xc20a881040, 0x5186cc0, 0xc208056090, 0x33, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/client.go:219 +0x4fc
net/http.(*Client).send(0xc20af961a8, 0xc20a881040, 0x33, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/client.go:142 +0x15b
net/http.(*Client).doFollowingRedirects(0xc20af961a8, 0xc20a881040, 0x4bf8370, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/client.go:367 +0xb25
net/http.(*Client).Do(0xc20af961a8, 0xc20a881040, 0xc20ad37df0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/client.go:177 +0x192
github.com/huin/goupnp/soap.(*SOAPClient).PerformAction(0xc20af96140, 0x4a6ced0, 0x2e, 0x49b2f10, 0x11, 0x46a8100, 0xc20859a2a0, 0x0, 0x0, 0x0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/huin/goupnp/soap/soap.go:51 +0x58a
github.com/huin/goupnp/dcps/internetgateway1.(*WANIPConnection1).DeletePortMapping(0xc2085fb080, 0x0, 0x0, 0x765f, 0xc20ad37d6b, 0x3, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/huin/goupnp/dcps/internetgateway1/internetgateway1.go:2583 +0x2a5
github.com/ethereum/go-ethereum/p2p/nat.(*upnp).DeleteMapping(0xc209844cf0, 0x495bed0, 0x3, 0x765f, 0x765f, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/nat/natupnp.go:95 +0xb0
github.com/ethereum/go-ethereum/p2p/nat.(*autodisc).DeleteMapping(0xc20800a2c0, 0x495bed0, 0x3, 0x765f, 0x765f, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:218 +0xc1
github.com/ethereum/go-ethereum/p2p/nat.func·001()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:106 +0x344
github.com/ethereum/go-ethereum/p2p/nat.Map(0x5187d58, 0xc20800a2c0, 0xc20a94cba0, 0x495bed0, 0x3, 0x765f, 0x765f, 0x49ca790, 0x12)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:117 +0x3ad
created by github.com/ethereum/go-ethereum/p2p/discover.newUDP
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:220 +0x2ce

goroutine 98 [runnable]:
github.com/ethereum/go-ethereum/eth.(*Ethereum).syncDatabases(0xc20aa34dc0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/backend.go:564 +0x53f
created by github.com/ethereum/go-ethereum/eth.(*Ethereum).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/backend.go:542 +0x1a5

goroutine 99 [runnable]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).txBroadcastLoop(0xc209dc5140)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:509 +0x99
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:152 +0x144

goroutine 100 [runnable]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).minedBroadcastLoop(0xc209dc5140)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:498 +0x88
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:155 +0x264

goroutine 101 [runnable]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).syncer(0xc209dc5140)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/sync.go:138 +0x23f
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:158 +0x281

goroutine 102 [runnable]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).txsyncLoop(0xc209dc5140)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/sync.go:104 +0x5f9
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/handler.go:159 +0x29e

goroutine 103 [chan receive]:
github.com/ethereum/go-ethereum/cmd/utils.func·001()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:135 +0x305
created by github.com/ethereum/go-ethereum/cmd/utils.StartEthereum
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:143 +0x312

goroutine 104 [select]:
github.com/ethereum/go-ethereum/xeth.(*XEth).start(0xc20baf0410)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:138 +0x702
created by github.com/ethereum/go-ethereum/xeth.New
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:128 +0x5c0

goroutine 113 [runnable]:
github.com/ethereum/go-ethereum/event/filter.(*FilterManager).filterLoop(0xc20ac86840)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/eth_filter.go:90 +0x7f9
created by github.com/ethereum/go-ethereum/event/filter.(*FilterManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/eth_filter.go:47 +0x32

goroutine 106 [IO wait, 5 minutes]:
net.(*pollDesc).Wait(0xc20a954e60, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20a954e60, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20a954e00, 0x0, 0x51867e0, 0xc20a9537d8)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:419 +0x40b
net.(*UnixListener).AcceptUnix(0xc2085f8a20, 0xdeaddeaddeaddead, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/unixsock_posix.go:282 +0x56
github.com/ethereum/go-ethereum/rpc/comms.func·003()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_unix.go:61 +0x52
created by github.com/ethereum/go-ethereum/rpc/comms.startIpc
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_unix.go:74 +0x2aa

goroutine 107 [select]:
github.com/ethereum/go-ethereum/xeth.(*XEth).start(0xc20baf04e0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:138 +0x702
created by github.com/ethereum/go-ethereum/xeth.New
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:128 +0x5c0

goroutine 111 [select]:
github.com/ethereum/go-ethereum/eth/fetcher.(*Fetcher).loop(0xc20b1c4500)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/fetcher/fetcher.go:242 +0x1ff6
created by github.com/ethereum/go-ethereum/eth/fetcher.(*Fetcher).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/fetcher/fetcher.go:144 +0x32

goroutine 112 [runnable]:
github.com/ethereum/go-ethereum/event/filter.(*FilterManager).filterLoop(0xc20ac86740)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/eth_filter.go:90 +0x7f9
created by github.com/ethereum/go-ethereum/event/filter.(*FilterManager).Start
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/eth_filter.go:47 +0x32

goroutine 9171 [runnable, locked to thread]:
github.com/ethereum/go-ethereum/crypto/secp256k1._Cfunc_secp256k1_ecdsa_sign_compact(0xc20830c580, 0xc20a87e9b0, 0xc20830c5c0, 0x40064a0, 0xc20830c5e0, 0xc20a3f68ac, 0xc200000001)
	github.com/ethereum/go-ethereum/crypto/secp256k1/_obj/_cgo_gotypes.go:55 +0x43
github.com/ethereum/go-ethereum/crypto/secp256k1.Sign(0xc20830c580, 0x20, 0x20, 0xc20830c5c0, 0x20, 0x20, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/secp256k1/secp256.go:140 +0x2dc
github.com/ethereum/go-ethereum/crypto.Sign(0xc20830c580, 0x20, 0x20, 0xc209836990, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/crypto.go:201 +0x2da
github.com/ethereum/go-ethereum/p2p/discover.encodePacket(0xc209836990, 0x46b5d03, 0x4846420, 0xc20a87e960, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:429 +0x466
github.com/ethereum/go-ethereum/p2p/discover.(*udp).send(0xc20a950070, 0xc20859aa80, 0x4846403, 0x4846420, 0xc20a87e960, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:409 +0x75
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc20a950070, 0xb6c4904a9cc74164, 0x262fbd6e1e655b5e, 0x930edec5969e3ce7, 0xa7bb95e582581f8a, 0xc937a02b614816e7, 0xb8ab940f6e6686f9, 0x4fe79019d2d7b6c, 0xc0e6fa4af77f65d7, 0xc20859aa80, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:276 +0x2d7
github.com/ethereum/go-ethereum/p2p/discover.func·002()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:227 +0x165
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:240 +0x65d

goroutine 9105 [IO wait]:
net.(*pollDesc).Wait(0xc20a7a2b50, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a7a2b50, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a7a2af0, 0x0, 0x0, 0x5188f50, 0xc20b3b3220, 0xecd4d2726, 0x212807d9, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a7a2af0, 0x6c52320, 0x0, 0x6c52320, 0xc20859a270, 0xecd4d2726, 0x212807d9, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20859a200, 0x6c52320, 0x0, 0x6c52320, 0xc20859a270, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20859a270, 0xecd4d2726, 0x212807d9, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20859a270, 0xecd4d2726, 0x212807d9, 0x4f91b40, 0xc20ad37d30, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20b3b3200, 0x11, 0x0, 0x0, 0x6c52290, 0xc20859a270, 0xecd4d2726, 0x212807d9, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2726, 0xe212807d9, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20859a270, 0xc20bad0e50, 0xecd4d2726, 0xe212807d9, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20b3b3200, 0x11, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20a237600, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 8747 [IO wait]:
net.(*pollDesc).Wait(0xc20a459bf0, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a459bf0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a459b90, 0x0, 0x0, 0x5188f50, 0xc20a126440, 0xecd4d2718, 0x34846749, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a459b90, 0x6c52320, 0x0, 0x6c52320, 0xc20a14f470, 0xecd4d2718, 0x34846749, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20a14f400, 0x6c52320, 0x0, 0x6c52320, 0xc20a14f470, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20a14f470, 0xecd4d2718, 0xc234846749, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20a14f470, 0xecd4d2718, 0x34846749, 0x4f91b40, 0xc209c9b4a0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20a126420, 0x12, 0x0, 0x0, 0x6c52290, 0xc20a14f470, 0xecd4d2718, 0x34846749, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2718, 0xe34846749, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20a14f470, 0xc20a8bfe50, 0xecd4d2718, 0xe34846749, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20a126420, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20aa14a00, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9118 [runnable]:
net.(*pollDesc).Wait(0xc20a7a2d10, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a7a2d10, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a7a2cb0, 0x0, 0x0, 0x5188f50, 0xc20b3b3340, 0xecd4d2735, 0x212f3f4d, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a7a2cb0, 0x6c52320, 0x0, 0x6c52320, 0xc20859a450, 0xecd4d2735, 0x212f3f4d, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20859a400, 0x6c52320, 0x0, 0x6c52320, 0xc20859a450, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20859a450, 0xecd4d2735, 0x212f3f4d, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20859a450, 0xecd4d2735, 0x212f3f4d, 0x4f91b40, 0xc20ad37e90, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc2085f61b7, 0x12, 0x0, 0x0, 0x6c52290, 0xc20859a450, 0xecd4d2735, 0x212f3f4d, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2735, 0xe212f3f4d, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20859a450, 0xc20a6d1b08, 0xecd4d2735, 0xe212f3f4d, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20800a500, 0x495ab70, 0x3, 0xc2085f61b7, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
net.*Dialer.Dial·fm(0x495ab70, 0x3, 0xc2085f61b7, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:38 +0x79
net/http.(*Transport).dial(0xc208056090, 0x495ab70, 0x3, 0xc2085f61b7, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:479 +0x84
net/http.(*Transport).dialConn(0xc208056090, 0x0, 0xc2085f61b0, 0x4, 0xc2085f61b7, 0x12, 0xc20a237600, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:564 +0x1678
net/http.func·019()
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:520 +0x42
created by net/http.(*Transport).getConn
	/usr/local/Cellar/go/1.4.2/libexec/src/net/http/transport.go:522 +0x335

goroutine 9104 [IO wait]:
net.(*pollDesc).Wait(0xc20a942ed0, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a942ed0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a942e70, 0x0, 0x0, 0x5188f50, 0xc20a631a60, 0xecd4d2726, 0x2127d665, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a942e70, 0x6c52320, 0x0, 0x6c52320, 0xc20859f440, 0xecd4d2726, 0x2127d665, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20859f400, 0x6c52320, 0x0, 0x6c52320, 0xc20859f440, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20859f440, 0xecd4d2726, 0xc22127d665, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20859f440, 0xecd4d2726, 0x2127d665, 0x4f91b40, 0xc209cf1d30, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20a631a40, 0x14, 0x0, 0x0, 0x6c52290, 0xc20859f440, 0xecd4d2726, 0x2127d665, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2726, 0xe2127d665, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20859f440, 0xc20bad1e50, 0xecd4d2726, 0xe2127d665, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20a631a40, 0x14, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20a2375f0, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9129 [runnable, locked to thread]:
github.com/ethereum/go-ethereum/crypto/secp256k1._Cfunc_secp256k1_ecdsa_sign_compact(0xc20ae7b0a0, 0xc209bf79f0, 0xc20ae7b0e0, 0x40064a0, 0xc20ae7b100, 0xc2080af8ac, 0xc200000001)
	github.com/ethereum/go-ethereum/crypto/secp256k1/_obj/_cgo_gotypes.go:55 +0x43
github.com/ethereum/go-ethereum/crypto/secp256k1.Sign(0xc20ae7b0a0, 0x20, 0x20, 0xc20ae7b0e0, 0x20, 0x20, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/secp256k1/secp256.go:140 +0x2dc
github.com/ethereum/go-ethereum/crypto.Sign(0xc20ae7b0a0, 0x20, 0x20, 0xc209836990, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/crypto.go:201 +0x2da
github.com/ethereum/go-ethereum/p2p/discover.encodePacket(0xc209836990, 0x46b5d03, 0x4846420, 0xc209bf79a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:429 +0x466
github.com/ethereum/go-ethereum/p2p/discover.(*udp).send(0xc20a950070, 0xc208380240, 0x4846403, 0x4846420, 0xc209bf79a0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:409 +0x75
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc20a950070, 0x7059f2ee308d03e9, 0xc8859a8acb4ceb8a, 0x2a4e94278afdd56a, 0xb9d8486ac7a2c43a, 0xbecb378a20b4dc4a, 0x948cc2df4f790e, 0x2e9a183968d1f1b5, 0x28c9fd47876f9fe7, 0xc208380240, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:276 +0x2d7
github.com/ethereum/go-ethereum/p2p/discover.func·002()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:227 +0x165
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:240 +0x65d

goroutine 8952 [IO wait]:
net.(*pollDesc).Wait(0xc20a908bc0, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a908bc0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a908b60, 0x0, 0x0, 0x5188f50, 0xc20a5ad7a0, 0xecd4d271d, 0x38a3c907, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a908b60, 0x6c52320, 0x0, 0x6c52320, 0xc20843ed50, 0xecd4d271d, 0x38a3c907, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20843ed00, 0x6c52320, 0x0, 0x6c52320, 0xc20843ed50, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20843ed50, 0xecd4d271d, 0xc238a3c907, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20843ed50, 0xecd4d271d, 0x38a3c907, 0x4f91b40, 0xc20a952d60, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20a5ad780, 0x12, 0x0, 0x0, 0x6c52290, 0xc20843ed50, 0xecd4d271d, 0x38a3c907, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d271d, 0xe38a3c907, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20843ed50, 0xc20a20ae50, 0xecd4d271d, 0xe38a3c907, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20a5ad780, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20bb0a380, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9103 [IO wait]:
net.(*pollDesc).Wait(0xc20a942fb0, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a942fb0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a942f50, 0x0, 0x0, 0x5188f50, 0xc20a631ae0, 0xecd4d2726, 0x212cf4f4, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a942f50, 0x6c52320, 0x0, 0x6c52320, 0xc20859f4a0, 0xecd4d2726, 0x212cf4f4, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20859f400, 0x6c52320, 0x0, 0x6c52320, 0xc20859f4a0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20859f4a0, 0xecd4d2726, 0xc2212cf4f4, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20859f4a0, 0xecd4d2726, 0x212cf4f4, 0x4f91b40, 0xc209cf1da0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20a631ac0, 0x11, 0x0, 0x0, 0x6c52290, 0xc20859f4a0, 0xecd4d2726, 0x212cf4f4, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2726, 0xe212cf4f4, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20859f4a0, 0xc209358e50, 0xecd4d2726, 0xe212cf4f4, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20a631ac0, 0x11, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20a2375e0, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9154 [chan receive]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup(0xc208015600, 0x856deed027be3104, 0xe6fb1020a7a7b357, 0x706282c1640f58c, 0x8bb80105e3ed930d, 0x77e531c02f5821e5, 0x6cfc13fbe44cfa84, 0xd9379ffa6660f2db, 0x1fcd7f2cca2f89dd, 0x0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:248 +0x757
github.com/ethereum/go-ethereum/p2p.(*discoverTask).Do(0xc20a69c280, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:238 +0x2a1
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9102 [IO wait]:
net.(*pollDesc).Wait(0xc20a942f40, 0x77, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20a942f40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).connect(0xc20a942ee0, 0x0, 0x0, 0x5188f50, 0xc20a631aa0, 0xecd4d2726, 0x212b4b8b, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:114 +0x24b
net.(*netFD).dial(0xc20a942ee0, 0x6c52320, 0x0, 0x6c52320, 0xc20859f470, 0xecd4d2726, 0x212b4b8b, 0x4f91b40, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:139 +0x37a
net.socket(0x495ab70, 0x3, 0x2, 0x1, 0x0, 0xc20859f400, 0x6c52320, 0x0, 0x6c52320, 0xc20859f470, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/sock_posix.go:91 +0x422
net.internetSocket(0x495ab70, 0x3, 0x6c52320, 0x0, 0x6c52320, 0xc20859f470, 0xecd4d2726, 0xc2212b4b8b, 0x4f91b40, 0x1, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/ipsock_posix.go:137 +0x148
net.dialTCP(0x495ab70, 0x3, 0x0, 0xc20859f470, 0xecd4d2726, 0x212b4b8b, 0x4f91b40, 0xc209cf1d70, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/tcpsock_posix.go:156 +0x125
net.dialSingle(0x495ab70, 0x3, 0xc20a631a80, 0x12, 0x0, 0x0, 0x6c52290, 0xc20859f470, 0xecd4d2726, 0x212b4b8b, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:235 +0x3f5
net.func·016(0xecd4d2726, 0xe212b4b8b, 0x4f91b40, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:163 +0x12c
net.dial(0x495ab70, 0x3, 0x6c52290, 0xc20859f470, 0xc20bacfe50, 0xecd4d2726, 0xe212b4b8b, 0x4f91b40, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:40 +0x6f
net.(*Dialer).Dial(0xc20acd95c0, 0x495ab70, 0x3, 0xc20a631a80, 0x12, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.4.2/libexec/src/net/dial.go:170 +0x4be
github.com/ethereum/go-ethereum/p2p.(*dialTask).Do(0xc20a2375c0, 0xc20b232300)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/dial.go:210 +0x1f4
github.com/ethereum/go-ethereum/p2p.func·008()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x53
created by github.com/ethereum/go-ethereum/p2p.func·009
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x58e

goroutine 9155 [runnable, locked to thread]:
github.com/ethereum/go-ethereum/crypto/secp256k1._Cfunc_secp256k1_ecdsa_sign_compact(0xc20a69d0c0, 0xc2096c9cc0, 0xc20a69d100, 0x40064a0, 0xc20a69d120, 0xc2097a58ac, 0xc200000001)
	github.com/ethereum/go-ethereum/crypto/secp256k1/_obj/_cgo_gotypes.go:55 +0x43
github.com/ethereum/go-ethereum/crypto/secp256k1.Sign(0xc20a69d0c0, 0x20, 0x20, 0xc20a69d100, 0x20, 0x20, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/secp256k1/secp256.go:140 +0x2dc
github.com/ethereum/go-ethereum/crypto.Sign(0xc20a69d0c0, 0x20, 0x20, 0xc209836990, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/crypto/crypto.go:201 +0x2da
github.com/ethereum/go-ethereum/p2p/discover.encodePacket(0xc209836990, 0x46b5d03, 0x4846420, 0xc2096c9c20, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:429 +0x466
github.com/ethereum/go-ethereum/p2p/discover.(*udp).send(0xc20a950070, 0xc20858c150, 0x4846403, 0x4846420, 0xc2096c9c20, 0x0, 0x0)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:409 +0x75
github.com/ethereum/go-ethereum/p2p/discover.(*udp).findnode(0xc20a950070, 0xa0cf7e6ac68fcf9f, 0x52bdee7f95c3f18e, 0xc94f77002979f03e, 0x1d76960a27f2531d, 0x6b5fd88f83af9855, 0xb9b9c81e1ce9ac08, 0xd05206d202a6ec73, 0xaf138fe2db18c9e4, 0xc20858c150, ...)
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:276 +0x2d7
github.com/ethereum/go-ethereum/p2p/discover.func·002()
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:227 +0x165
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup
	/Users/justin/Desktop/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:240 +0x65d
```

--- a/core/chain_manager.go
+++ b/core/chain_manager.go
@@ -76,7 +76,8 @@ type ChainManager struct {
 	cache        *lru.Cache // cache is the LRU caching
 	futureBlocks *lru.Cache // future blocks are blocks added for later processing
 
-	quit chan struct{}
+	quit    chan struct{}
+	running int32 // running must be called automically
 	// procInterrupt must be atomically called
 	procInterrupt int32 // interrupt signaler for block processing
 	wg            sync.WaitGroup
@@ -443,6 +444,9 @@ func (bc *ChainManager) setTotalDifficulty(td *big.Int) {
 }
 
 func (bc *ChainManager) Stop() {
+	if !atomic.CompareAndSwapInt32(&bc.running, 0, 1) {
+		return
+	}
 	close(bc.quit)
 	atomic.StoreInt32(&bc.procInterrupt, 1)
 

